### DISCUSSION TITLE
using command search (ctrl+C) to get menu path - then opening that path

### DISCUSSION THREAD

**Discussion by kenfhill84083** [freeplane/freeplane#1263] [Help]
There have been a few times when I've been in the Command Search dialog and ready to click one of the items when I've thought, "I wonder what state that feature is in".  (I am using CS throughout every session, working almost exclusively in full-screen mode.  So CS is my Freeplane best friend.)

For example, "show note icons" or "notes follow selection".  I can find out by by playing around with the map or actually traversing the menus.  It is not onerous but sometimes it is a bit tedious.  And it can distract me from what I'm trying to do.

I am wondering if I can take a menu path I get from Command Search Ctrl+C and somehow use it to actually open that menu to look at the status of a feature.  One path I copied in CS is: "View->Notes->Note follows selection".  Could I use a script to get to "View > Notes" to check the status?  Or is there a way other than the CS-path approach?

I have been learning scripting but am at the very beginning of the journey and I am a slow learner.  Is there anything you could point me to in the API that might help?

Thanks.

![freeplane _ going from command search to open menu](https://github.com/freeplane/freeplane/assets/6380301/5c07f43f-1af6-451f-8f23-e871c9c42864)

**Comment by amariusz** [comment]
Having state preview directly in Command Search would be nice. It's probably a good idea to consider within a feature request.

**Reply by dpolivaev** [reply]
I can add a state preview to the dialog. However, the state would be not updated if the item is shown and the state is changed outside.

**Reply by kenfhill84083** [reply]
That would be fantastic.  I generally use Command Search and then close it until needed again.  Thanks.

**Reply by macmarrum** [reply]
Cool. I'm looking forward to it :eyes:

**Comment by euu2021** [comment]
I'm happy to listen that you are learning scripting. It really takes the experience to another level.

In that situation that you mention, one idea is to make the on/off state appear in the status line, so you don't need to navigate to the menu. Another option is showing it in a dialog that pops up.

PS: searching the API I see that there are no methods to get this kind of information. So, I think that only an advanced script would work, because it needs internal methods to get the state of those menu items.

**Reply by kenfhill84083** [reply]
Thanks for the research.  I find the API very cryptic.

Maybe scripting can keep my mind from deteriorating - like sudoku - unless my head explodes first.

**Reply by euu2021** [reply]
That's an interesting project to learn scripting: a Sudoku generator for FreePlane.

> I find the API very cryptic

Yes, I have a long term project of making it easier to understand. But, I'm still in the phase of understanding how the scripts actually work inside FP.

**Reply by dpolivaev** [reply]
I feel your pain: the API often is far from ideal.

**Comment by dpolivaev** [comment]
Status of selectable menu items and boolean preferences is shown in the command search dialog in the latest preview

**Comment by dpolivaev** [comment]
I am closing the discussion now, if you need anything else please start a new discussion

**Reply by kenfhill84083** [reply]
Thanks.  That is a great implementation.
